script/workLoggingSystem.js

```diff
@@ -136,9 +136,96 @@ $("#submitNewTask").click(function() {
 
-    createNewTask(title,des,priority,difficulty,creator,assignee,estimatedTime,status,category);
+    //editing
+    if(taskpriority(priority)  && taskdifficulty(difficulty)  && tasktime(estimatedTime)  && taskstatus(status)  && taskcategory(category))
+        {
+            console.log("working");
+            createNewTask(title,des,priority,difficulty,creator,assignee,estimatedTime,status,category);
+        }
+    else
+        console.log("wrong values");
+
 });
 
+function taskpriority(priority) {
+    if(priority == "High" || priority == "Medium"|| priority == "Low")
+        {   
+            document.getElementById("ppp").style.backgroundColor = "#ffc107";
+            document.getElementById("ppp").style.color = "black";
+            return true;
+        }
+    else    
+        {
+            document.getElementById("ppp").style.backgroundColor = "red";
+            document.getElementById("ppp").style.color = "#ffc107";
+            alert("Enter valid priority");
+            return false;
+        }
+}
+function taskdifficulty(difficulty) {
+    if(difficulty == "High" || difficulty == "Medium"|| difficulty == "Low")
+        {   
+            document.getElementById("ddd").style.backgroundColor = "#ffc107";
+            document.getElementById("ddd").style.color = "black";
+            return true;
+        }
+    else    
+        {
+            document.getElementById("ddd").style.backgroundColor = "red";
+            document.getElementById("ddd").style.color = "#ffc107";
+            alert("Enter valid difficulty");
+            return false;
+        }
+}
+function taskstatus(status) {
+    if(status == "Icebox" || status == "Ready to start"|| status == "Under Progress"|| status == "Blocked"|| status == "Completed")
+        {   
+            document.getElementById("sss").style.backgroundColor = "#ffc107";
+            document.getElementById("sss").style.color = "black";
+            return true;
+        }
+    else    
+        {
+            document.getElementById("sss").style.backgroundColor = "red";
+            document.getElementById("sss").style.color = "#ffc107"
+            alert("Enter valid status");
+            return false;
+        }
+    
+}
+function taskcategory(category) {
+    if(category == "Business" || category == "Development"|| category == "Marketting")
+        {   
+            document.getElementById("ccc").style.backgroundColor = "#ffc107";
+            document.getElementById("ccc").style.color = "black";
+            return true;
+        }
+    else 
+        {
+            document.getElementById("ccc").style.backgroundColor = "red";
+            document.getElementById("ccc").style.color = "#ffc107";
+            alert("Enter valid category");            
+            return false;
+        }   
+}
+function tasktime(estimatedTime) {
+    var a= parseInt(estimatedTime);
+    if(a > 0)
+    {
+        document.getElementById("ttt").style.backgroundColor = "#ffc107";
```

Comment:
Let's have just one single method and validate all as functions are costly and will increase the time complexity